### PR TITLE
[#291] Refactoring flow of displaying optional add-ons prompts

### LINKED ISSUE #291: Refactoring flow of displaying optional add-ons prompts
## Why

According to [this discussion](https://github.com/nimblehq/elixir-templates/discussions/261), the team agree to refactor the flow of add-ons prompts

## Who Benefits?

Developers

### PR DESCRIPTION
**Pull request by hanam1ni**
- Closes #291

## What happened 👀

Update optional addon prompt logic to display and collect responses from the user before applying any addon.

## Insight 📝

- Move functions in project helper to the project module.
- Update ExVCR checking logic in the Wallaby addon.

## Proof Of Work 📹

- Optional addon prompt is working properly

image

- Addons are applied

image

### CODE REVIEW

File: lib/nimble_template/projects/project.ex [hunk 1]
```diff
@@ -80,6 +80,14 @@ defmodule NimbleTemplate.Projects.Project do
     |> List.first()
   end
 
+  @spec prepend_optional_addon(__MODULE__.t(), atom()) :: __MODULE__.t()
+  def prepend_optional_addon(
+        %__MODULE__{optional_addons: existing_addons} = project,
+        addon_module
+      ) do
+    Map.put(project, :optional_addons, [addon_module | existing_addons])
+  end
```
Comment (nvminhtue) [inline lib/nimble_template/projects/project.ex:89]:
```suggestion
      ), do: Map.put(project, :optional_addons, [addon_module | existing_addons])
```

File: lib/nimble_template/templates/variants/mix/template.ex [hunk 2]
```diff
@@ -6,6 +6,10 @@ defmodule NimbleTemplate.Templates.Mix.Template do
   alias NimbleTemplate.Addons
   alias NimbleTemplate.Projects.Project
 
+  def pre_apply(%Project{} = project) do
+    install_addon_prompt(project, Addons.Mimic)
+  end
```
Comment (nvminhtue) [inline lib/nimble_template/templates/variants/mix/template.ex:11]:
```suggestion
  def pre_apply(%Project{} = project), do:
    install_addon_prompt(project, Addons.Mimic)
```

**Review by nvminhtue — APPROVED** [review]
some minor comments, the rest is LGTM

File: lib/nimble_template/templates/variants/mix/template.ex (hunk 2 above)
Comment (hanam1ni) [reply lib/nimble_template/templates/variants/mix/template.ex]: Updated in [9782e2e](https://github.com/nimblehq/elixir-templates/pull/292/commits/9782e2ef5fef654f7a75a5bf77f6729d46406c43)

File: lib/nimble_template/projects/project.ex (hunk 1 above)
Comment (hanam1ni) [reply lib/nimble_template/projects/project.ex]: Updated in [9782e2e](https://github.com/nimblehq/elixir-templates/pull/292/commits/9782e2ef5fef654f7a75a5bf77f6729d46406c43)

**Review by Nihisil — APPROVED** [review]

**Pull request merged**